### PR TITLE
Update: QualityTaks to 2.0.0

### LINKED ISSUE #416: [Model Update]: bamm:io.catenax.quality_task:1.0.0 to 2.0.0 with SAMM
## Update Reason
The use case quality group decided to upgrade this model to version 2.0.0: 
- Use new versions of ESMF toolings
- Convert from BAMM to SAMM
- Make adaptions in model structure

## MS1 Criteria

- [X] The model that should be updated exists
- [X ] The referenced use case exists: It is use case CX QAX
- [X] The potential updates are discussed with all stakeholders
- [X] A decision on versioning was made in consensus with the affected parties [according to the guidelines](https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/documentation/GOVERNANCE.md#versioning).
--> new version will be 2.0.0
- [ ] The potential update will be either 
  - [ ] backward compatible or 
  - [X] a (chain of) migration strategy(-ies) from all non-deprecated previous versions to the new model will be developed
- [x] A modeling steward has been assigned

### PR DESCRIPTION
**Pull request by TorstenHirsch**
## Description

Converting the model to SAMM.
### Added
- integration of the shared BPNL trait
- integration of shared e-Mail trait

### Changed
- changed all properties to optional except one unique identifier per entity
- change property cxbpn cxBusinessPartnerNumber

### Removed

Closes #416 

## MS2 Criteria
(to be filled out by PR reviewer)
- [x] the model **validates** with the SAMM SDS SDK in the version specified in the Readme.md of this repository by the time of the MS2 check  (e.g., 'java -jar samm-cli.jar aspect \<path-to-aspect-model\> validate ). The  SAMM CLI is available [here](https://eclipse-esmf.github.io/esmf-developer-guide/tooling-guide/samm-cli.html) and in [GitHub](https://github.com/eclipse-esmf/esmf-sdk/releases/tag/v2.3.2)
- [x] use **Camel-Case** (e.g., "MyModelElement" or "TimeDifferenceGmtId", when in doubt follow https://google.github.io/styleguide/javaguide.html#s5.3-camel-case)
- [x] the identifiers for all model elements **start with a capital letter** except for properties
- [x] the identifier for **properties starts with a small letter**
- [x] all model elements **at least contain the fields "preferred name" and "description"** in English language. The description must be comprehensible. It is not required to write full sentences but style should be consistent over the whole model
- [x] Property and the referenced Characteristic should not have the same name
- [x] the versioning in the URN **follows semantic versioning**, where minor version bumps are backwards compatible and major version bumps are not backwards compatible. 
- [x] use **abbreviations only when necessary** and if these are sufficiently common
- [x] **avoid redundant prefixes in property names** (consider adding properties to an enclosing Entity or even adapt the namespace of the model elements, e.g., instead of having two properties `DismantlerId` and `DismantlerName` use an Entity `Dismantler` with the properties `name` and `id` or use a URN like `io.catenax.dismantler:0.0.1`)
- [x] fields `preferredName` and `description` are not the same
- [x] **`preferredName` should be human readable** and follow normal orthography (e.g., no camel case but normal word separation)
- [x] name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.
- [x] units are referenced from the SAMM unit catalog whenever possible
- [x] **use constraints** to make known constraints from the use case explicit in the aspect model 
- [x] when relying on **external standards**, they are referenced through a **"see"** element
- [x] all properties with an [simple type](https://eclipse-esmf.github.io/samm-specification/2.0.0/datatypes.html) have an example value
- [x] metadata.json exists with status "release"
- [x] generated json schema validates against example json payload
- [x] file RELEASE_NOTES.md exists and contains entries for proposed model changes 
- [x] all contributors to this model are mentioned in copyright header of model file

## MS3 Criteria
(to be filled out by semantic modeling team before merge to main-branch)
- [x] All required reviewers have approved this PR (see reviewers section)
- [x] The new aspect (version) will be implemented by at least one data provider
- [x] The new aspect (version) will be consumed by at least one data consumer
- [x] There exists valid test data
- [x] In case of a new (incompatible) major version to an existing version, a migration strategy has been developed
- [x] The model has at least version '1.0.0'
- [x] If a previous model exists, model deprecation has been checked for previous model
- [x] The release date in the Release Note is set to the date of the MS3 approval

### CODE REVIEW

**Comment by TorstenHirsch** [comment]
modelling_team

**Comment by TorstenHirsch** [comment]
@agg3fe

File: io.catenax.quality_task/2.0.0/QualityTask.ttl [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+:cxBusinessPartnerNumber a samm:Property ;
+   samm:preferredName "Catena-X Business partner number"@en ;
+   samm:description "The Catena-X business partner number for this company."@en ;
+   samm:characteristic ext-number:BpnlTrait ;
+   samm:exampleValue "BPNL000000000123" .
+
+:name a samm:Property ;
+   samm:preferredName "Company name"@en ;
+   samm:description "Name of the comapny"@en ;
```
Comment (agg3fe) [inline io.catenax.quality_task/2.0.0/QualityTask.ttl:181]: typo `comapny`

File: io.catenax.quality_task/2.0.0/QualityTask.ttl [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+   samm:description "This entity groups all meta information properties. Meta information properties give hints for a Catena-X data consumer to better understand, how this data set was created on the Catena-X data provider side."@en ;
+   samm:properties ( :selectionCriteria [ samm:property :selectionStart; samm:optional true ] [ samm:property :selectionEnd; samm:optional true ] ) .
+
+:recordStatus a samm:Property ;
+   samm:preferredName "Record status"@en ;
+   samm:description "The record operation enunumeration can be used to realize delta update concept.\nDelta updateupdate concept means:\n- You transfer an initial load of data\n- After the first week only the delta to the initial load is transferred\n\n\nThe record status describes whether this record is:\n- new=>This record is transferred the first time\n- update=> Some properties of this record have changed compared to a previous transfer\n- delete=> This record was transferred in the initial load or in a previous delta update, but is not \nused any more and therefore it should be deleted on data consumer side\n- same=> This record was transferred in the initial load or in a previous delta update"@en ;
```
Comment (agg3fe) [inline io.catenax.quality_task/2.0.0/QualityTask.ttl:71]: typo `enunumeration` , `updateupdate` ,` \nused`

File: io.catenax.quality_task/2.0.0/QualityTask.ttl [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+   samm:characteristic :MetaCharacteristic .
+
+:ListOfQualityTasks a samm-c:List ;
+   samm:preferredName "List of quality tasks"@en ;
+   samm:description "The list of quality tasks itself."@en ;
+   samm:dataType :SingleQualityTask .
+
+:MetaCharacteristic a samm:Characteristic ;
+   samm:preferredName "MetaCharacteristic"@en ;
+   samm:description "Chracteristic for the meta information."@en ;
```
Comment (agg3fe) [inline io.catenax.quality_task/2.0.0/QualityTask.ttl:56]: typo `Chracteristic`

File: io.catenax.quality_task/2.0.0/QualityTask.ttl [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+@prefix ext-uuid: <urn:samm:io.catenax.shared.uuid:1.0.0#> .
+
+:QualityTask a samm:Aspect ;
+   samm:preferredName "QualityTasks"@en ;
+   samm:description "A list of quality tasks (qTask). "@en ;
+   samm:properties ( :listOfQualityTasks [ samm:property :metaInformation; samm:optional true ] ) ;
+   samm:operations ( ) ;
+   samm:events ( ) .
+
+:listOfQualityTasks a samm:Property ;
```
Comment (agg3fe) [inline io.catenax.quality_task/2.0.0/QualityTask.ttl:39]: Property and the referenced Characteristic should not have the same name.

**Comment by agg3fe** [comment]
@TorstenHirsch we have one clause: 'name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.'
In your case, I think. you have only one property in other models also, right? then We can keep the model name as plural and don't need to make it singular.
For this model, for now we can keep it like this or if you want you can revert it to plural name if you want.

File: io.catenax.quality_task/2.0.0/QualityTask.ttl (hunk 2 above)
Comment (agg3fe) [reply io.catenax.quality_task/2.0.0/QualityTask.ttl:71]: @TorstenHirsch looks like this one is skipped.

**Comment by TorstenHirsch** [comment]
> @TorstenHirsch we have one clause: 'name of aspect is singular except if it only has one property which is a Collection, List or Set. In theses cases, the aspect name is plural.' In your case, I think. you have only one property in other models also, right? then We can keep the model name as plural and don't need to make it singular. For this model, for now we can keep it like this or if you want you can revert it to plural name if you want.

@agg3fe  Let's keep the model name qualityTask - if possible

File: io.catenax.quality_task/2.0.0/QualityTask.ttl (hunk 4 above)
Comment (TorstenHirsch) [reply io.catenax.quality_task/2.0.0/QualityTask.ttl:39]: Fixed - please check

File: io.catenax.quality_task/2.0.0/QualityTask.ttl (hunk 1 above)
Comment (TorstenHirsch) [reply io.catenax.quality_task/2.0.0/QualityTask.ttl]: fixed

File: io.catenax.quality_task/2.0.0/QualityTask.ttl (hunk 2 above)
Comment (TorstenHirsch) [reply io.catenax.quality_task/2.0.0/QualityTask.ttl:71]: fixed

File: io.catenax.quality_task/2.0.0/QualityTask.ttl [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+   samm:description "This entity groups all meta information properties. Meta information properties give hints for a Catena-X data consumer to better understand, how this data set was created on the Catena-X data provider side."@en ;
+   samm:properties ( :selectionCriteria [ samm:property :selectionStart; samm:optional true ] [ samm:property :selectionEnd; samm:optional true ] ) .
+
+:recordStatus a samm:Property ;
+   samm:preferredName "Record status"@en ;
+   samm:description "The record operation enunumeration can be used to realize delta update concept.\nDelta update concept means:\n- You transfer an initial load of data\n- After the first week only the delta to the initial load is transferred\n\n\nThe record status describes whether this record is:\n- new=>This record is transferred the first time\n- update=> Some properties of this record have changed compared to a previous transfer\n- delete=> This record was transferred in the initial load or in a previous delta update, but is not \nused any more and therefore it should be deleted on data consumer side\n- same=> This record was transferred in the initial load or in a previous delta update"@en ;
```
Comment (agg3fe) [inline io.catenax.quality_task/2.0.0/QualityTask.ttl:71]: please fix typo enunumeration. should be enumeration.

**Comment by agg3fe** [comment]
@TorstenHirsch please fix one typo, then we are good for MS2.

**Comment by TorstenHirsch** [comment]
> @TorstenHirsch please fix one typo, then we are good for MS2.

Done - let's go for MS2 :-)

**Pull request merged**